### PR TITLE
Propose to use express-interceptor instead of express-hijackresponse

### PR DESCRIPTION
**Pull request by fer8a**
Hello there,

The idea behind this PR is to replace express-hijackresponse dependency for another module called express-interceptor, it allows you to define a previous step before sending a response and do whatever you need with it. Also you can avoid calling next() over and over and overall it helps to have a cleaner base code.

### CODE REVIEW

File: lib/compiless.js [hunk 1]
```diff
@@ -6,34 +6,60 @@ var Path = require('path'),
...
+                }
+
+                delete req.headers['if-modified-since']; // Prevent false positive conditional GETs after enabling compiless
+
                 var contentType = res.getHeader('Content-Type'),
                     matchContentType = contentType && contentType.match(/^text\/less(?:;\s*charset=([a-z0-9\-]+))?$/i);
+
                 // The mime module doesn't support less yet, so we fall back:
-                if (matchContentType || (/\.less(?:\?.*)?$/.test(req.url) && contentType === 'application/octet-stream')) {
+                if (matchContentType || contentType === 'application/octet-stream') {
```
Comment (Munter) [inline lib/compiless.js]: There was probably a reason for this check

Comment (fer8a) [inline lib/compiless.js]: Yes indeed! that is checking that the file is actually a .less file

Now this validation is being added to the function isInterceptable from express-interceptor. So if it's not a .less file the response just gets ignore.

**Comment by papandreou** [comment]
+1, I've been waiting for someone to come up with an alternative to express-hijackresponse, whose existence is keeping me awake at night.

**Comment by p1nox** [comment]
Is this ready to merge?

**Comment by papandreou** [comment]
Sorry for the lack of feedback. I tried one of my applications against your branch, but I ran into a regression where unrelated requests never receive a response. I reduced it to a simple failing test that I landed on the master branch here: https://github.com/papandreou/express-compiless/commit/f8c45d2720bb6a52d7528ea447b6ccf75b0bd374

If you rebase on that and make it work, I'll take another look :)

Also, please squash the first two commits. It's doesn't really make sense to keep the `package.json` change separate from the code change. The first commit leaves the repo in a state where nothing works, which isn't `git bisect`-friendly.

**Comment by papandreou** [comment]
Rewrote the test suite to use unexpected-express on master. I hope that makes it a bit nicer to work with.

**Comment by p1nox** [comment]
@fer8a can you take a look here? :)

**Comment by fer8a** [comment]
Yes, as soon as I get home I'll rebase/squash as needed and re-test everything.

**Comment by p1nox** [comment]
Awesome, we will make some changes in the package, but that won't affect the api :+1:.

**Comment by fer8a** [comment]
Thank you @papandreou unexpected-express was indeed very helpful. I did some changes and moved the condition to check for the less file inside the intercept function and set isInterceptable to return true.

**Comment by papandreou** [comment]
Thanks, it works much better now!

**Comment by papandreou** [comment]
Released in 2.1.0.

**Comment by p1nox** [comment]
Awesome! Thanks to both @fer8a and @papandreou :feelsgood:

**Pull request merged**